src/include/cart/types.h

```diff
@@ -20,6 +20,17 @@
 
 #include <stdint.h>
 #include <gurt/types.h>
+
+/**
+ * Protocol info used to query list of protocols and devices.
+*/
+struct crt_protocol_info {
```

Comment:
by our convention we typedef external struct types to 'struct_name'_t  so  crt_protocol_info_t

Reply:
It's in the C style guide to not use typedefs actually:
> Do not use typedef for new declarations, as this hides the type of a field for very little benefit. Never typedef pointers, as the * makes C pointer declarations obvious. Hiding it inside a typedef just obfuscates the code.

Reply:
@ashleypittman since you are the main maintainer of the style guide, can you please confirm? :)

Reply:
I can confirm that a large part of our public API is in contradiction to our style guidelines.

Personally my preferred style is what you've done here as I find that typedefs only serve to obfuscate which is exactly what the coding style says and I didn't even write that part.